File: exercise/c#/Day24/Day24/Position.cs
namespace Day24
{
    public record Position(int Horizontal, int Depth)
    {
        public Position ChangeDepth(int newDepth) => this with {Depth = newDepth};
        public Position MoveHorizontally(int newHorizontal) => this with {Horizontal = newHorizontal};
    }
}

File: exercise/ts/day24/src/position.ts
export class Position {
    constructor(public readonly horizontal: number, public readonly depth: number) {
    }

    public changeDepth(newDepth: number): Position {
        return new Position(this.horizontal, newDepth);
    }

    public moveHorizontally(newHorizontal: number): Position {
        return new Position(newHorizontal, this.depth);
    }
}